ext/session/session.c

```diff
@@ -1686,15 +1700,56 @@ static PHP_FUNCTION(session_set_cookie_params)
 		RETURN_FALSE;
 	}
 
-	convert_to_string_ex(lifetime);
+	if (Z_TYPE_P(lifetime_or_options) == IS_ARRAY) {
+		int found = 0;
+		zend_string *key;
+		zval *value;
+
+		ZEND_HASH_FOREACH_STR_KEY_VAL(Z_ARRVAL_P(lifetime_or_options), key, value) {
+			if (key) {
+				ZVAL_DEREF(value);
+				if(!strcasecmp("lifetime", ZSTR_VAL(key))) {
+					lifetime = value;
+					convert_to_string_ex(lifetime);
+					found++;
+				} else if(!strcasecmp("path", ZSTR_VAL(key))) {
+					path = Z_STR_P(value);
+					found++;
+				} else if(!strcasecmp("domain", ZSTR_VAL(key))) {
+					domain = Z_STR_P(value);
+					found++;
+				} else if(!strcasecmp("secure", ZSTR_VAL(key))) {
+					secure = zval_is_true(value);
+					secure_null = 0;
+					found++;
+				} else if(!strcasecmp("httponly", ZSTR_VAL(key))) {
+					httponly = zval_is_true(value);
+					httponly_null = 0;
+					found++;
+				} else if(!strcasecmp("samesite", ZSTR_VAL(key))) {
+					samesite = Z_STR_P(value);
+					found++;
+				}
```

Comment:
Should an unknown option generate a warning?

Reply:
I've added the warning and the error. Thanks